addons/point_of_sale/migrations/13.0.1.0.1/post-migration.py

```diff
@@ -86,6 +86,37 @@ def delete_pos_order_line_with_empty_order_id(env):
     )
 
 
+def create_pos_picking_types(env):
+    """Create picking types according to former pos.config locations
+    """
+    openupgrade.logged_query(
+        env.cr, """
+        SELECT pc.id, {stock_location_id}
+        FROM pos_config pc
+        JOIN stock_picking_type spt ON pc.picking_type_id = spt.id
+        WHERE {stock_location_id} != spt.default_location_src_id
+        """.format(
+            stock_location_id=openupgrade.get_legacy_name('stock_location_id')
+        )
+    )
+    # Unzip and zip again so we browse all the configs at once
+    try:
+        config_ids, location_ids = zip(*env.cr.fetchall())
+    except ValueError:
+        # No values to unpack so no results were fetched
+        return
+    for config, location_id in zip(
+        env["pos.config"].browse(config_ids),
+        location_ids
+    ):
+        config.picking_type_id = config.picking_type_id.copy({
```

Comment:
Only do this if the location is not the same as the one in the picking type.

Reply:
I'm already filtering them in the sql